torch/nn/utils/parametrizations.py

```diff
@@ -1,10 +1,273 @@
 import torch
+from torch import Tensor
 from ..utils import parametrize
 from ..modules import Module
 from .. import functional as F
 
 from typing import Optional
 
+
+def _is_orthogonal(Q, eps=None):
+    n, k = Q.size(-2), Q.size(-1)
+    Id = torch.eye(k, dtype=Q.dtype, device=Q.device)
+    # A reasonable eps, but not too large
+    eps = 10. * n * torch.finfo(Q.dtype).eps
+    return torch.allclose(Q.transpose(-2, -1).conj() @ Q, Id, atol=eps)
+
+
+def make_orthogonal(A):
+    """ Assume that A is a tall matrix.
+    Compute the Q factor s.t. A = QR (A may be complex) and diag(R) is real and non-negative
+    """
+    X, tau = torch.geqrf(A)
+    Q = torch.linalg.householder_product(X, tau)
+    # The diagonal of X is the diagonal of R (which is always real) so we normalise by its signs
+    Q *= X.diagonal(dim1=-2, dim2=-1).sgn().unsqueeze(-2)
+    return Q
+
+
+class _Orthogonal(Module):
+    base: Tensor
+
+    def __init__(self, weight, parametrization: str = None, *, extra_memory=True) -> None:
+        super().__init__()
+
+        # Note [Householder complex]
+        # For complex tensors, it is not possible to compute the tensor `tau` necessary for
+        # linalg.householder_product from the reflectors.
+        # To see this, note that the reflectors have a shape like:
+        # 0 0 0
+        # * 0 0
+        # * * 0
+        # which, for complex matrices, give n(n-1) (real) parameters. Now, you need n^2 parameters
+        # to parametrize the unitary matrices. Saving tau on its own does not work either, because
+        # not every combination of `(A, tau)` gives a unitary matrix, meaning that if we optimise
+        # them as independent tensors we would not maintain the constraint
+        # An equivalent reasoning holds for rectangular # matrices
```

Comment:
What's the meaning of the # here?

Reply:
When you join two consecutive comment lines in vim with `J`, it does not remove the `#` and I forgot to delete it :D